src/profile/ProfileForm.tsx

```diff
@@ -37,585 +25,405 @@ interface ChangedUser {
+                      }
+                      onClick={() => {
+                        setShowPassword((show) => !show);
+                      }}
+                    >
+                      {showPassword ? (
+                        <FontAwesomeIcon icon={faEyeSlash} />
+                      ) : (
+                        <FontAwesomeIcon icon={faEye} />
+                      )}
+                    </Button>
                   }
-                  value={state.changedUser?.password ?? ""}
                 />
-                <InputGroup.Append>
-                  <Button
-                    variant={showPassword ? "outline-primary" : "outline-secondary"}
-                    onClick={() => {
-                      setShowPassword((s) => !s);
-                    }}
-                  >
-                    <FontAwesomeIcon icon={showPassword ? faEyeSlash : faEye} />
-                  </Button>
-                </InputGroup.Append>
-              </InputGroup>
-              <Form.Control
-                type="hidden"
-                isInvalid={
-                  state.errors.password && state.errors.password.length > 0
-                }
-                value={state.changedUser?.password}
+              </Col>
+            </Form.Row>
+          </Col>
+        </Form.Group>
+        {passwordConfirmRequired && (
+          <>
+            <hr />
+            <Form.Group>
+              <Form.Label>Confirm with your password to save:</Form.Label>
+              <Field
+                errors={props.context.errors}
+                type="password"
+                name="cur_password"
+                ref={props.context.register({
+                  required: passwordConfirmRequired
+                    ? "USER.PASSWORD.CURRENT_REQUIRED"
+                    : undefined,
+                })}
+                onChange={() => {
+                  props.context.triggerValidation("cur_password");
+                }}
               />
-              <Form.Control.Feedback type="invalid">
-                <ul>
-                  {state.errors.password?.map((msg) => (
-                    <li key={msg}>{ERRORS[msg]}</li>
-                  ))}
-                </ul>
-              </Form.Control.Feedback>
-            </Col>
-          </Form.Row>
-        </Col>
-      </Form.Group>
-
-      {renderFooter ? (
-        renderFooter(
-          state.isLoading,
-          state.isChecking || Object.values(state.errors).flat().length > 1
-        )
-      ) : (
-        <Button
-          type="submit"
-          disabled={
-            state.isLoading ||
-            state.isChecking ||
-            Object.values(state.errors).flat().length > 1
-          }
-        >
-          {state.isLoading ? "Saving..." : "Save"}
-        </Button>
-      )}
-    </Form>
+            </Form.Group>
+          </>
+        )}
+        <NonFieldErrors errors={generalErrors || []} />
```

Comment:
General errors should be moved to the top of the form.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/b614bfaf1fda88049bccda144a4f1230dcd1743e